RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT_vect.cc

```diff
@@ -0,0 +1,1030 @@
+      double w = tks.pi_[i] * tks.errsum_[i];
+      sumwz += w * tks.z_[i];
+      sumwt += w * tks.t_[i];
+      sumw += w;
+    }
+    y.z_[k] = sumwz / sumw;
+    y.t_[k] = sumwt / sumw;
+    
+    // estimate Tcrit, eventually do this in the same loop
+    double a = 0, b = 0;
+    for (unsigned int i = 0; i < nt; i++) {
+      double dx = tks.z_[i] - y.z_[k];
+      double dt = tks.t_[i] - y.t_[k];
+      double w = tks.pi_[i] * tks.errsum_[i];
+      a += w * std::pow(dx, 2) * tks.dz2_[i] + std::pow(dt,2) * tks.dt2_[i];
+      b += w;
+    }
+    double Tc = 2. * a / b; // the critical temperature of this vertex
+    if (Tc > T0) T0 = Tc;
+  }// vertex loop (normally there should be only one vertex at beta=0)
+  
+#ifdef VI_DEBUG
+  if(verbose_){
+    std::cout << "DAClustrizerInZT_vect.beta0:   Tc = " << T0 << std::endl;
+    int coolingsteps =  1 - int(std::log(T0 * betamax) / std::log(coolingFactor_));
+    std::cout << "DAClustrizerInZT_vect.beta0:   nstep = " << coolingsteps << std::endl;
+  }
+#endif
+
+
+  if (T0 > 1. / betamax) {
+    return betamax / std::pow(coolingFactor_, int(std::log(T0 * betamax) / std::log(coolingFactor_)) - 1);
+  } else {
+    // ensure at least one annealing step
+    return betamax * coolingFactor_;
+  }
+}
+
+
+  
+bool 
+DAClusterizerInZT_vect::split(const double beta,  track_t &tks, vertex_t & y, double threshold ) const{
+  // split only critical vertices (Tc >~ T=1/beta   <==>   beta*Tc>~1)
+  // an update must have been made just before doing this (same beta, no merging)
+  // returns true if at least one cluster was split
+  
+  constexpr double epsilonz=1e-3;      // minimum split size z
+  constexpr double epsilont=1e-2;      // minimum split size t
+  unsigned int nv = y.getSize();
+  
+  // avoid left-right biases by splitting highest Tc first
+  
+  std::vector<std::pair<double, unsigned int> > critical;
+  for(unsigned int k=0; k<nv; k++){    
+    double Tc= 2*y.swE_[k]/y.sw_[k];
+    if (beta*Tc > threshold){
+      critical.push_back( make_pair(Tc, k));
+    }
+  }
+  if (critical.empty()) return false;
+
+
+  std::stable_sort(critical.begin(), critical.end(), std::greater<std::pair<double, unsigned int> >() );
+  
+  
+  bool split=false;
+  const unsigned int nt = tks.getSize();
+
+  for(unsigned int ic=0; ic<critical.size(); ic++){
+    unsigned int k=critical[ic].second;
+
+    // estimate subcluster positions and weight
+    double p1=0, z1=0, t1=0, w1=0;
+    double p2=0, z2=0, t2=0, w2=0;
+    for(unsigned int i=0; i<nt; ++i){
+      if (tks.Z_sum_[i] > 1.e-100) {
+	// winner-takes-all, usually overestimates splitting
+	double tl = tks.z_[i] < y.z_[k] ? 1.: 0.;
+	double tr = 1. - tl;
+
+	// soften it, especially at low T
+	/*
+	double arg = ( tks.z_[i] - y.z_[k] ) * sqrt(beta * tks.dz2_[i]); // + std::fabs(tks.t_[i] - y.t_[k])
+	std::cout << arg << std::endl;
+	if(std::fabs(arg) < 20){
+	  double t = local_exp(-arg);
+	  tl = t/(t+1.);
+	  tr = 1/(t+1.);
+	}
+	*/
```

Comment:
should this be left configurable?
This was added recently (5 mo) to the Z_vect version.
If there is a good reason to drop, perhaps this should be picked up there as well.
I'd like to minimize functional divergence between the two.